core/src/main/scala/org/apache/spark/metrics/source/StaticSources.scala

```diff
@@ -60,3 +60,32 @@ object CodegenMetrics extends Source {
   val METRIC_GENERATED_METHOD_BYTECODE_SIZE =
     metricRegistry.histogram(MetricRegistry.name("generatedMethodSize"))
 }
+
+/**
+ * :: Experimental ::
+ * Metrics for access to the hive external catalog.
+ */
+@Experimental
+object HiveCatalogMetrics extends Source {
```

Comment:
Ah, the reason you can't is because to register this source it needs to be in the list above. This made me realize I forgot to add it to the list actually: https://github.com/VideoAmp/spark-public/pull/6